### PR TITLE
basic devise setup #2

### PR DESCRIPTION
**Pull request by dewanhimanshu**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -8,7 +8,8 @@
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
-
+  <p class="notice"><%= notice %></p>
```
Comment (sushantahujaa) [inline app/views/layouts/application.html.erb:13]: This should be under body tags

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '1f8576d21f22e0953f38ca39ce4b85dbed406e9e3ed1086436f84925a6c5c9d8ae44cf7be287f43c62dbbdab2ee611776700e318b37c15cf2ab80a903706c42e'
```
Comment (sushantahujaa) [inline config/initializers/devise.rb]: This line should be uncommented

File: test/models/user_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (sushantahujaa) [inline test/models/user_test.rb]: Test cases should be in rspec

File: test/fixtures/users.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (sushantahujaa) [inline test/fixtures/users.yml:1]: Use rspec

File: test/application_system_test_case.rb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-require "test_helper"
+require 'test_helper'
```
Comment (sushantahujaa) [inline test/application_system_test_case.rb]: Use rspec-rails

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (sushantahujaa) [inline app/models/user.rb]: Let's not do database_authenticable for now

**Review by sushantahujaa — COMMENTED** [review]
@dewanhimanshu Check my comments, Ping me if you need anything. After removing database_authenticable, some files will be removed. Please check that

File: config/initializers/devise.rb (hunk 2 above)
Comment (dewanhimanshu) [reply config/initializers/devise.rb]: ok

File: test/models/user_test.rb (hunk 3 above)
Comment (dewanhimanshu) [reply test/models/user_test.rb]: but we havent added rspec in our project till now

File: app/models/user.rb (hunk 6 above)
Comment (dewanhimanshu) [reply app/models/user.rb]: i have removed this

File: test/application_system_test_case.rb (hunk 5 above)
Comment (dewanhimanshu) [reply test/application_system_test_case.rb]: i have changes this

**Pull request merged**